src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Extensions.cs

```diff
@@ -0,0 +1,539 @@
+
+                            foreach (var extension in declaredBaseExtension.BaseExtensionsNoUseSiteDiagnostics)
+                            {
+                                if (extension.DeclaringCompilation != this.DeclaringCompilation)
+                                {
+                                    extension.AddUseSiteInfo(ref useSiteInfo);
+                                }
+                            }
+                            diagnostics.Add(Locations[0], useSiteInfo);
+                        }
+                    }
+
+                    return result.ToImmutableAndFree();
+                }
+            }
+        }
+
+        private ExtensionInfo GetDeclaredExtensionInfo()
+        {
+            Debug.Assert(this.IsExtension);
+
+            if (ReferenceEquals(_lazyDeclaredExtensionInfo, ExtensionInfo.Sentinel))
+            {
+                BindingDiagnosticBag diagnostics = BindingDiagnosticBag.GetInstance();
+
+                var original = Interlocked.CompareExchange(ref _lazyDeclaredExtensionInfo,
+                    this.MakeDeclaredExtensionInfo(basesBeingResolved: null, diagnostics),
+                    ExtensionInfo.Sentinel);
+
+                if (ReferenceEquals(original, ExtensionInfo.Sentinel))
+                {
+                    AddDeclarationDiagnostics(diagnostics);
+                }
+
+                diagnostics.Free();
+            }
+
+            return _lazyDeclaredExtensionInfo;
+        }
+
+        /// <summary> Bind the base extensions for all parts of an extension.</summary>
+        private ExtensionInfo MakeDeclaredExtensionInfo(ConsList<TypeSymbol>? basesBeingResolved, BindingDiagnosticBag diagnostics)
+        {
+            Debug.Assert(this.IsExtension);
+
+            bool? anyExplicit = null;
+
+            bool sawUnderlyingType = false;
+            bool reportedUnderlyingTypeConflict = false;
+            TypeSymbol? underlyingType = null;
+            SourceLocation? underlyingTypeLocation = null;
+
+            Debug.Assert(basesBeingResolved == null || !basesBeingResolved.ContainsReference(this.OriginalDefinition));
+            var newBasesBeingResolved = basesBeingResolved.Prepend(this.OriginalDefinition);
+            var baseExtensionsBuilder = ArrayBuilder<NamedTypeSymbol>.GetInstance();
+            var baseExtensionLocations = SpecializedSymbolCollections.GetPooledSymbolDictionaryInstance<NamedTypeSymbol, SourceLocation>();
+
+            foreach (var declaration in this.declaration.Declarations)
+            {
+                ExtensionInfo one = MakeOneDeclaredExtensionInfo(newBasesBeingResolved, declaration, diagnostics, out bool sawPartUnderlyingType);
+                sawUnderlyingType |= sawPartUnderlyingType;
+
+                if (anyExplicit == null)
+                {
+                    anyExplicit = one.IsExplicit;
+                }
+                else if (anyExplicit != one.IsExplicit)
```

Comment:
>else if (anyExplicit != one.IsExplicit) [](http://example.com/codeflow?start=16&length=39)

The logic here looks suspicious because ```explicit``` is treated differently than ```implicit```. What exactly is our intent here? Let's say we have the following parts: implicit, explicit, implicit. Are we going to report two errors, one for the second and one for the third part? But if we have have: explicit, implicit, explicit, then only one error (for the second part) will be reported? 
Why not simply grabbing the value from the first part and checking all the other parts against that value? Whatever the first part says goes as the result, no overwriting.

Reply:
`explicit` is more restrictive, so I'm treating the presence of any `explicit` to mean the extension is explicit in aggregate.
That said, I'm okay with just taking the modifier from the first part.